### PR TITLE
Lint expectations

### PR DESCRIPTION
**Pull request by dottorblaster**
As above, linting expectations. YAY!

### CODE REVIEW

File: src/dsl/validation.rs [hunk 1]
```diff
@@ -116,11 +158,58 @@ mod tests {
...
         let json_schema = get_json_schema();
-        let validation_result = validate(&json_value, "156F64", &json_schema);
+        let validation_result = validate(&json_value, "156F64", &json_schema, &engine);
 
         assert_eq!(validation_result.is_ok(), true);
     }
+
+    #[test]
+    fn validate_invalid_expectations_check() {
+        let input = r#"
```
Comment (arbulu89) [inline src/dsl/validation.rs:174]: Maybe we could remove most of this boilerplate test data, only leaving the needed fields. In this case only the expectations and values part are important.
Just to reduce the number of lines hehe
But well, if you want to have all the data for consistency is fine too

File: src/dsl/validation.rs [hunk 2]
```diff
@@ -27,7 +30,42 @@ pub fn validate(
...
+    };
+
+    let (_, expression_errors): (Vec<_>, Vec<_>) = json_check
+        .get("expectations")
+        .unwrap_or(&json!([]))
+        .as_array()
+        .unwrap_or(&Vec::new())
+        .iter()
+        .map(|value| {
+            let expect = value.get("expect").unwrap().as_str().unwrap();
```
Comment (arbulu89) [inline src/dsl/validation.rs:45]: We are missing the `expect_same` scenario here.
We should get one or the other.

File: src/dsl/validation.rs [hunk 3]
```diff
@@ -27,7 +30,42 @@ pub fn validate(
...
     };
 
-    validation_result
+    let mut schema_validation_errors = match validation_result {
+        Ok(_) => vec![],
+        Err(errors) => errors,
+    };
+
+    let (_, expression_errors): (Vec<_>, Vec<_>) = json_check
+        .get("expectations")
```
Comment (arbulu89) [inline src/dsl/validation.rs:39]: We would need to repeat the same thing with the `values` entry, as we do the same rhai evaluation there

**Review by arbulu89 — CHANGES_REQUESTED** [review]
@dottorblaster Looks super promising.
2 minor changes requested. Validate the `values` entry and include `expect_same`.
After that, we are ready to merge

**Comment by dottorblaster** [comment]
@nelsonkopliku it actually did! :smile: Rebasing made the additional test I wrote pass in a breeze

File: src/dsl/validation.rs (hunk 2 above)
Comment (dottorblaster) [reply src/dsl/validation.rs]: Right :+1:

**Review by arbulu89 — APPROVED** [review]
Thanks @dottorblaster ,
without understanding much about all of this list operations, the domain logic looks perfect to me

**Comment by dottorblaster** [comment]
@arbulu89 it's just a lot of resilient `map` statements :sweat_smile:

**Comment by dottorblaster** [comment]
@arbulu89 anyway you might find useful to have a look at this technique, I love that and I find it very useful to collect errors

https://doc.rust-lang.org/rust-by-example/error/iter_result.html

**Pull request merged**